syntax = "proto3";

option java_package = "ex.grpc";
option objc_class_prefix = "HSW";

package device;

service TemperatureService {
  rpc set_temperature (Temperature) returns (Empty) {}
  rpc get_temperature (Device) returns (stream Temperature) {}
}

service UsageService {
  rpc latest_electricity_usage (ElectricityUsage)  returns (Empty) {}
  rpc latest_gas_usage (GasUsage) returns (Empty) {}
}

message Empty {
}

message Device {
   string device = 1;
}

message Temperature {
  string device = 1;
  float celcius = 2;  
}

message ElectricityUsage {
  string device = 1;
  int64 kw = 2;
}

message GasUsage {
  string device = 1;
  int64 m3 = 2;
}

